### ISSUE TITLE
Recording metrics for webaction response statusCode in user-events service. 

### ISSUE THREAD

**Issue by neerajmangal** [apache/openwhisk#4718]
A developer can specify the status code using `statusCode` in response, in case of web actions. 
The top-level activation response captures the activation `statusCode` as 0,1,2, and 3 based on `success`, `application error`, `action developer error` and `whisk internal error`. 

If action developer sending a `statusCode` explicitly in response, it will be stored in activation's response result as `statusCode` (like below as HTTP status code `401` ) 

```
"response": {
        "status": "application error",
        "statusCode": 0,
        "success": false,
        "result": {
            "error": {
                "body": "Authentication credentials are invalid.",
                "statusCode": 401
            }
        }
    }
```

Capturing metrics for the statusCode returned by the action developer in user-events can give insights like
"How many requests failed with which error code".  

Conditions - 

As returning a proper status code, is the responsibility of the developer, if a developer does not specify a return `statusCode` in response the metric will not be present/recoded. 

Please let me know your thoughts. 

PS: - While doing some tests, I see top-level `statusCode` always as 0 in case of all error scenarios.

**Comment by rabbah** [comment]
The status code in the response is orthogonal to the webaction http status code. It is an internal code only. Changing this is an api change as the value 0-3 are part of the activation record schema.

**Comment by neerajmangal** [comment]
> The status code in the response is orthogonal to the webaction http status code. It is an internal code only. Changing this is an api change as the value 0-3 are part of the activation record schema.

Yup, you are right, I confused myself with activation response vs end-user response. 
I see the status code in web actions HTTP status code which makes sense.

```
< HTTP/1.1 401 Unauthorized
< X-Request-ID: ZPyOucUbIuUz2eqskYfGTqJrSROxBOHP
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: OPTIONS, GET, DELETE, POST, PUT, HEAD, PATCH
< Access-Control-Allow-Headers: Authorization, Origin, X-Requested-With, Content-Type, Accept, User-Agent
< x-openwhisk-activation-id: 85daf1091d3940e49af1091d3950e47d
< Server: akka-http/10.1.8
< Date: Fri, 08 Nov 2019 04:21:37 GMT
< Content-Type: text/html; charset=UTF-8
< Content-Length: 39
```

**Comment by neerajmangal** [comment]
Closed by #4726
